### PR TITLE
Add remaining UCI app config parameters to nocli uwb range start

### PR DESCRIPTION
**Pull request by corbin-phipps**
### Type

- [ ] Bug fix
- [ ] Feature addition
- [x] Feature update
- [ ] Breaking change
- [ ] Non-functional change
- [ ] Documentation
- [ ] Infrastructure

### Goals

This PR adds the remaining UCI app config params to nocli. This includes CLI options, some additional validation (not all), and additional code to print out parameters.

### Technical Details

- Add new CLI options
- Add to and improve parameter display code in ProcessApplicationConfigurationParameters
- Add validation for ResultReportConfig
- Improve GetValueMap()
- Remove redundant ParameterTypesVariant
- Update some comments in FiraDevice.hxx

### Test Results

nocli output looks correct

### Reviewer Focus

Apologies for the long PR, but there are so many UCI app config parameters!

Also, in this line: `oss << "0x" << std::setw(2) << std::internal << std::setfill('0') << std::hex << +value << " ";`, I had to add `"0x"` instead of `std::showbase` because it interferes with `std::setw(2)` and `std::setfill('0')`, since the added base is considered part of the width, thus truncating the added 0.

### Future Work

Additional validation for the newly added parameters.

I also really don't like how CLI11 adds "UINT" or "ENUM" when displaying the options. Perhaps this can be changed.

### Checklist

- [x] Build target `all` compiles cleanly.
- [x] clang-format and clang-tidy deltas produced no new output.
- [x] Newly added functions include doxygen-style comment block.

### CODE REVIEW

**Comment by corbin-phipps** [comment]
Output:
image

**Comment by corbin-phipps** [comment]
image

image

**Comment by abeltrano** [comment]
> I also really don't like how CLI11 adds "UINT" or "ENUM" when displaying the options. Perhaps this can be changed.

💯 I've never liked it.

**Comment by abeltrano** [comment]
> ### Future Work
I'm also hoping we can make some of the primary parameters positional such that switches don't always need to be provided. This is mostly for commands that only take 1 or 2 arguments.

File: tools/cli/NearObjectCli.cxx [hunk 1]
```diff
@@ -483,6 +545,40 @@ NearObjectCli::AddSubcommandUwbRangeStart(CLI::App* parent)
             m_cliData->appConfigParamsData.destinationMacAddresses.value().insert(uwb::UwbMacAddress::FromString(m_cliData->destinationMacAddressString, macAddressType).value());
         }
 
+        // Parse and validate ResultReportConfig
+        constexpr int resultReportConfigurationSize = 4; // TODO: Find a way to calculate this number
```
Comment (abeltrano) [inline tools/cli/NearObjectCli.cxx:549]:
```suggestion
        constexpr int resultReportConfigurationSize = magic_enum::enum_count<ResultReportConfig>();
```

File: tools/cli/NearObjectCli.cxx [hunk 2]
```diff
@@ -483,6 +545,40 @@ NearObjectCli::AddSubcommandUwbRangeStart(CLI::App* parent)
...
+                    return false;
+                }
+            }
+            return true;
+        };
+
+        if (IsValidResultReportConfigurationString(m_cliData->resultReportConfigurationString)) {
+            m_cliData->appConfigParamsData.resultReportConfig.emplace();
+
+            const std::bitset<4> resultReportConfigurationBits(m_cliData->resultReportConfigurationString);
```
Comment (abeltrano) [inline tools/cli/NearObjectCli.cxx:567]: Should this be `resultReportConfigurationSize` instead?
```suggestion
            const std::bitset<resultReportConfigurationSize > resultReportConfigurationBits(m_cliData->resultReportConfigurationString);
```

File: tools/cli/NearObjectCli.cxx [hunk 3]
```diff
@@ -483,6 +545,40 @@ NearObjectCli::AddSubcommandUwbRangeStart(CLI::App* parent)
...
+            if (resultReportConfigurationBits[0]) {
```
Comment (abeltrano) [inline tools/cli/NearObjectCli.cxx:568]: FYI, `std::bitset` also has a `test()` function is does the same thing as the subscript operator here. I like that `test()` more directly indicates what's being done, but I'm mostly bringing this up for informational purposes. Feel free to close.

**Review by abeltrano — APPROVED** [review]
Thanks for doing this, I know it must have been tedious 😒

**Pull request merged**